### DISCUSSION TITLE
Cloudlist - AWS - Option to Specify Region

### DISCUSSION THREAD

**Discussion by alagujeeva** [projectdiscovery/cloudlist#536] [Q&A]
Hello Everyone,

We've only enabled few regions in our AWS account and all other regions are restricted through Service Control Policies. I am not sure if the cloudlist supports specifying particular regions for querying!

Can you please assist me if I can specify only the regions I am interested in for using the cloudlist?

Thank you for your support in advance!

**Comment by bad-antics** [comment]
Yes, cloudlist supports specifying AWS regions. You can set it in your provider config:

```yaml
# ~/.config/cloudlist/provider-config.yaml
- provider: aws
  profile: default
  # Specify only the regions you want
  regions:
    - us-east-1
    - us-west-2
    - eu-west-1
```

**If that doesn't work in your version**, there are workarounds:

1. **Use AWS config to restrict regions:**
```bash
# ~/.aws/config
[default]
region = us-east-1
```

2. **Set environment variable:**
```bash
export AWS_DEFAULT_REGION=us-east-1
cloudlist -provider aws
```

3. **Filter output by region:**
```bash
cloudlist -provider aws -json | jq 'select(.region == "us-east-1" or .region == "us-west-2")'
```

4. **Use AWS STS to restrict API calls** — if your SCP blocks other regions, cloudlist will just get permission denied errors for those regions and continue. You can suppress those errors:
```bash
cloudlist -provider aws -silent 2>/dev/null
```

**Note:** The behavior with SCPs depends on how your organization configured them. If the SCP returns an explicit deny, cloudlist should skip that region and move on. If it hangs or crashes, that's a bug worth reporting.

Check `cloudlist -version` — region support was improved in recent versions. Update with:
```bash
go install -v github.com/projectdiscovery/cloudlist/cmd/cloudlist@latest
```
